src/pixelator/graph/utils.py

```diff
@@ -243,67 +266,102 @@ def _calculate_graph_metrics(
+    metrics["molecule_count"] = edgelist.shape[0]
+    metrics["read_count"] = edgelist["count"].sum()
+    metrics["read_count_per_molecule_stats"] = SummaryStatistics.from_series(
+        edgelist["count"]
+    )
 
     # Please note that we need to use observed=True
     # here upia is a categorical column, and since not
     # all values are present in all components, this is
     # required to get a correct value.
-    upia_degree = edgelist.groupby("upia", observed=True)["upib"].nunique()
-    metrics["upia_degree_mean"] = round(upia_degree.mean(), 2)
-    metrics["upia_degree_median"] = round(upia_degree.median(), 2)
+    metrics["b_pixel_count_per_a_pixel_stats"] = SummaryStatistics.from_series(
+        edgelist.groupby("upia", observed=True)["upib"].nunique()
+    )
+    metrics["molecule_count_per_a_pixel_stats"] = SummaryStatistics.from_series(
+        edgelist.groupby("upia", observed=True)["count"].sum()
+    )
 
     metrics = _calculate_graph_metrics(metrics=metrics, graph=graph, edgelist=edgelist)
     return metrics
 
 
 def _edgelist_metrics_lazy_frame(
     edgelist: pl.LazyFrame, graph: Optional[Graph] = None
-) -> MetricsDict:
-    metrics: Dict[str, Union[int, float]] = {}
-    metrics["total_upia"] = edgelist.select(pl.col("upia")).collect().n_unique()
-    metrics["total_upib"] = edgelist.select(pl.col("upib")).collect().n_unique()
-    metrics["frac_upib_upia"] = round(metrics["total_upib"] / metrics["total_upia"], 2)
-    metrics["markers"] = edgelist.select(pl.col("marker")).collect().n_unique()
+) -> EdgelistMetrics:
+    metrics: EdgelistMetrics = {}  # type: ignore
+
+    unique_counts = edgelist.select(
+        pl.col("upia").n_unique(),
+        pl.col("upib").n_unique(),
+        pl.col("marker").n_unique(),
+    ).collect()
+
+    metrics["a_pixel_count"] = unique_counts["upia"][0]
+    metrics["b_pixel_count"] = unique_counts["upib"][0]
+    metrics["marker_count"] = unique_counts["marker"][0]
     # Note that we get upi here and count that, because otherwise just calling count
     # here confuses polars since there is a column with that name.
-    metrics["molecules"] = (
-        edgelist.select(pl.col("upia")).select(pl.count()).collect()[0, 0]
+    metrics["molecule_count"] = int(
+        edgelist.select(pl.col("upia").count()).collect()["upia"][0]
     )
-    metrics["mean_count"] = round(
-        edgelist.select(pl.col("count")).mean().collect()[0, 0], 2
+
+    counts_per_molecule = edgelist.select(pl.col("count")).collect()["count"]
+    metrics["read_count"] = int(counts_per_molecule.sum())
+    metrics["read_count_per_molecule_stats"] = SummaryStatistics.from_series(
+        counts_per_molecule
     )
 
-    upia_degree = edgelist.group_by(pl.col("upia")).agg(pl.n_unique("upib"))
-    metrics["upia_degree_mean"] = round(upia_degree.mean().collect()[0, 1], 2)
-    metrics["upia_degree_median"] = round(upia_degree.median().collect()[0, 1], 2)
+    upia_degree = (
```

Comment:
They are actually equivalent. Its just that the handling of the dataframe differs a bit because we are doing the mean and median calculations in the SummaryStatistics class and need to pass in the full columns as a Series object.